pandas/core/frame.py

```diff
@@ -113,7 +113,15 @@
     axes_single_arg="{0 or 'index', 1 or 'columns'}",
     optional_by="""
         by : str or list of str
-            Name or list of names which refer to the axis items.""",
+            Name or list of names matching axis levels or off-axis labels.
+
+            - if `axis` is 0 or `'index'` then `by` may contain index
+              levels and/or column labels
+            - if `axis` is 1 or `'columns'` then `by` may contain column
+              levels and/or index labels
+
+            Support for specify index/column levels was added in
```

Comment:
This could be a `..versionmodified:`

    .. versionmodified:: 0.22.0
       Allow specifying index or column level names.

Reply:
done